.vsts.pipelines/jobs/ci-linux_bootstrap.yml

```diff
@@ -0,0 +1,106 @@
+    artifactName: ${{ format('{0} $(type)', parameters.job) }}
+    # Use ":z" to set selinux flag for sharing in build-owned root dir. https://docs.docker.com/storage/bind-mounts/#configure-the-selinux-label
+    docker.agentSrc.map: -v $(Build.SourcesDirectory):/agentSrc:z
+    docker.agentSrc.work: -w /agentSrc
+    docker.drop.map: -v $(dropDirectory):/drop:z
+    docker.logs.map: -v $(dropDirectory)/logs:/logs:z
+    docker.root.map: -v $(rootDirectory):/root:z
+    docker.run: docker run --rm
+    docker.src.map: -v $(rootDirectory)/sb/source-build:/src:z
+    docker.src.work: -w /src
+    docker.tb.map: -v $(rootDirectory)/sb/tarball:/tb:z
+    docker.tb.work: -w /tb
+    docker.bst.map: -v $(System.ArtifactsDirectory)/boostrap_root:/bst:z
+    docker.bst.work: -w /bst
+    dropDirectory: $(stagingDirectory)/drop
+    imageName: ${{ parameters.imageName }}
+    reportPrebuiltLeaks: ${{ parameters.reportPrebuiltLeaks }}
+    generatePrebuiltBurndown: ${{ parameters.generatePrebuiltBurndown }}
+    rootDirectory: $(Build.SourcesDirectory)/..
+    stagingDirectory: $(rootDirectory)/sb/staging
+    tarballName: tarball_$(Build.BuildId)
+    SOURCE_BUILD_SKIP_SUBMODULE_CHECK: true
+    # Default type, can be overridden by matrix legs.
+    type: Production
+
+  steps:
+  # - template: ../steps/cleanup-unneeded-files.yml
+  - template: ../steps/docker-cleanup-linux.yml
+  - template: ../steps/calculate-config-flags.yml
+
+  # Create working directory and copy source into it.
+  - script: |
+      set -ex
+      df -h
+      $(docker.run) $(docker.root.map) $(docker.agentSrc.map) $(docker.agentSrc.work) $(imageName) bash -c '
+        rm -rf /root/sb/
+        mkdir -p /root/sb/tarball'
+      du -h $(rootDirectory) | sort -h | tail -n 50
+    displayName: Clean sb directory
+
+  - task: DownloadBuildArtifacts@0
+    displayName: Download CentOS artifacts
+    inputs:
+      buildType: 'current'
+      itemPattern: 'Tarball centos71 Offline/$(tarballName).tar.gz'
+      artifactName: 'Tarball centos71 Offline'
+      downloadPath: '$(System.ArtifactsDirectory)/boostrap_root'
+
+  - script: |
+      set -ex
+      df -h
+      $(docker.run) $(docker.bst.map) $(docker.bst.work) $(imageName) bash -c '
+      git clone https://github.com/dotnet/source-build-reference-packages'
+      du -h $(rootDirectory) | sort -h | tail -n 50
+    displayName: clone source-build-reference-packages
+    condition: always()
+    continueOnError: true
+  
+  - script: |
+      set -ex 
+      df -h 
+      $(docker.run) --network=none $(docker.bst.map) $(docker.bst.work) $(imageName) bash -c '
+      ls /bst
+      mkdir -p /bst/sb/tarball
+      tar -xvf "/bst/Tarball centos71 Offline/$(tarballName).tar.gz" -C /bst/sb/tarball --no-same-owner --strip-components=1
+      ls /bst/sb/tarball
+      /bst/sb/tarball/tool-bootstrapping/bootstrap.sh /bst/bootstrap_dir /bst/sb/tarball /bst/source-build-reference-packages'
```

Comment:
Were you planning on breaking this up into multiple steps and cleaning up between them?  Also, do you want to upload logs for each step as artifacts?